### PR TITLE
Add test for cpu snapshot

### PR DESCRIPTION
**Pull request by Yingshun**
This PR covers test to check if snapshot func works well for vm
with host-mode cpu mode.

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
(.libvirt-ci-venv-ci-runtest-2OsIRo) [root@hpe-dl120gen10-01 cpu]# avocado run --vt-type libvirt vcpu_misc..snapshot
JOB ID     : f047913c79565399c4d848a5ac1dcef1b4e3bc02
JOB LOG    : /root/avocado/job-results/job-2019-10-28T23.17-f047913/job.log
 (1/1) type_specific.io-github-autotest-libvirt.vcpu_misc.positive_test.snapshot: PASS (5.25 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 6.81 s
(.libvirt-ci-venv-ci-runtest-2OsIRo) [root@hpe-dl120gen10-01 cpu]#

**Comment by chunfuwen** [comment]
Can we use current code base or cfg  to accommodate this case?

**Comment by Yingshun** [comment]
> Can we use current code base or cfg to accommodate this case?

So far, python scripts and cfg files are divided by features.
Cases like checking snapshot with special cpu settings are not include by any file.

File: libvirt/tests/src/cpu/vcpu_misc.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        if test_operations:
+            for action in test_operations.split(","):
+                if action == "do_snapshot":
+                    do_snapshot(vm_name, expected_str_before_startup)
+
+        # Check if vm could start successfully
+        try:
+            result = virsh.start(vm_name, debug=True)
+            libvirt.check_exit_status(result)
+        except virt_vm.VMStartError as details:
```
Comment (chloerh) [inline libvirt/tests/src/cpu/vcpu_misc.py]: I think there will not be exceptions. virsh.xxx() sets ignore_status = True. check_exit function will raise exceptions.TestFail. Maybe you actually want to use 'vm.start()' ?

Comment (chloerh) [reply libvirt/tests/src/cpu/vcpu_misc.py]: I mean no 'virt_vm.VMStartError'

Comment (Yingshun) [reply libvirt/tests/src/cpu/vcpu_misc.py]: Thanks @chloerh  
Updated.

File: libvirt/tests/src/cpu/vcpu_misc.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        """
+        snapshot_name = vm_name + "-snap"
+        virsh_dargs = {'debug': True}
+
+        cmd_result = virsh.snapshot_create_as(vm_name, snapshot_name,
+                                              **virsh_dargs)
+        libvirt.check_exit_status(cmd_result)
+
+        try:
+            virsh.snapshot_list(vm_name, **virsh_dargs)
```
[the remark this answers is not in the document]
Comment (Yingshun) [reply libvirt/tests/src/cpu/vcpu_misc.py]: Updated.

**Pull request merged**